src/rgw/rgw_op.cc

```diff
@@ -2124,7 +2124,7 @@ int RGWGetObj::get_data_cb(bufferlist& bl, off_t bl_ofs, off_t bl_len)
 bool RGWGetObj::prefetch_data()
 {
   /* HEAD request, stop prefetch*/
-  if (!get_data) {
+  if (!get_data || s->info.env->exists("HTTP_AUTH")) {
```

Comment:
We cannot change the method because the client already signed the request using the Method we must send to the RGW the GET request as the user itself send it, that way we will not need to delegate access to the Nginx itself to authorize users

Reply:
i see. i guess that's also why it can't be a query param (ie ?auth) like other subresources, because those are part of the signature too. can we at least prefix the header with `x-rgw-` to make it clear that it's an rgw extension? maybe x-rgw-cache-auth?

Reply:
ack

Reply:
Yeah, sure, I will change it